test/e2e/sidecar_test.go

```diff
@@ -168,6 +168,11 @@ func getJaegerAgentAsSidecarDefinition(name, namespace string) *v1.Jaeger {
 		},
 		Spec: v1.JaegerSpec{
 			Strategy: v1.DeploymentStrategyAllInOne,
+			JaegerCommonSpec: v1.JaegerCommonSpec{
+				// add inject annotation to jaeger deployment to non existing Jaeger instance
+				// we don't want to inject agent to jaeger deployment because the port collisions
+				Annotations: map[string]string{inject.Annotation: "doesNotExists"},
```

Comment:
Not sure I get it. Are you saying that if a Jaeger CR for all-in-one is created, it will get a sidecar injected if the namespace is annotated, unless the deployment is annotated with a non-existing Jaeger instance?

I think the operator should handle this special case: we can detect that the deployment is a Jaeger deployment, skip the deployment.

Reply:
> Not sure I get it. Are you saying that if a Jaeger CR for all-in-one is created, it will get a sidecar injected if the namespace is annotated, unless the deployment is annotated with a non-existing Jaeger instance? 

Yes, it will get sidecar injected.

Reply:
I will create an issue for it

Reply:
https://github.com/jaegertracing/jaeger-operator/issues/900

Reply:
Would it be possible to fix it in this PR? This currently isn't a problem, as the Jaeger deployment has to be explicitly annotated to reach this situation, whereas current deployment will fail once this PR is merged and the namespace gets annotated. It should be relatively simple, as the `Needed` could just check for label `app: jaeger`, or `app.kubernetes.io/part-of: jaeger`.

Reply:
I would prefer a separate PR this is already a medium size.

Reply:
Alright, let's just keep in mind that we probably don't want to release 1.17 with this bug.